File: src/main/java/com/wikia/tibia/enums/BestiaryOccurrence.java
package com.wikia.tibia.enums;

import com.fasterxml.jackson.annotation.JsonValue;
import com.wikia.tibia.interfaces.Description;

public enum BestiaryOccurrence implements Description {

    COMMON("Common"),
    UNCOMMON("Uncommon"),
    RARE("Rare"),
    VERY_RARE("Very Rare");

    private final String description;

    BestiaryOccurrence(String description) {
        this.description = description;
    }

    @JsonValue
    public String getDescription() {
        return description;
    }
}

File: src/main/kotlin/com/wikia/tibia/enums/BestiaryOccurrence.kt
package com.wikia.tibia.enums

import com.fasterxml.jackson.annotation.JsonValue
import com.wikia.tibia.interfaces.Description

enum class BestiaryOccurrence(@JsonValue override val description: String) : Description {
  COMMON("Common"),
  UNCOMMON("Uncommon"),
  RARE("Rare"),
  VERY_RARE("Very Rare");
}
